samples/interactive-tutorials/src/main/java/product/ImportProductsBigQueryTable.java

```diff
@@ -0,0 +1,118 @@
+      String.format(
+          "projects/%s/locations/global/catalogs/default_catalog/" + "branches/default_branch",
+          PROJECT_ID);
+  private static final String DATASET_ID = "products";
+  private static final String TABLE_ID = "products";
+  // TO CHECK ERROR HANDLING USE THE TABLE WITH INVALID PRODUCTS:
+  // TABLE_ID = "products_some_invalid"
+  private static final String DATA_SCHEMA = "product";
+
+  public static void main(String[] args) throws IOException, InterruptedException {
+    importProductsFromBigQuery();
+  }
+
+  public static void importProductsFromBigQuery() throws IOException, InterruptedException {
+    // TRY THE FULL RECONCILIATION MODE HERE:
+    ReconciliationMode reconciliationMode = ReconciliationMode.INCREMENTAL;
+    ImportProductsRequest importBigQueryRequest =
+        getImportProductsBigQueryRequest(reconciliationMode);
+
+    try (ProductServiceClient serviceClient = ProductServiceClient.create()) {
+      String operationName =
+          ProductServiceClient.create()
+              .importProductsCallable()
+              .call(importBigQueryRequest)
+              .getName();
+      System.out.printf("OperationName = %s\n", operationName);
+
+      OperationsClient operationsClient = serviceClient.getOperationsClient();
+      Operation operation = operationsClient.getOperation(operationName);
+
+      while (!operation.getDone()) {
+        // Keep polling the operation periodically until the import task is done.
+        int awaitDuration = 30000;
+        Thread.sleep(awaitDuration);
+        operation = operationsClient.getOperation(operationName);
+      }
+
+      if (operation.hasMetadata()) {
+        ImportMetadata metadata = operation.getMetadata().unpack(ImportMetadata.class);
+        System.out.printf(
+            "Number of successfully imported products: %s\n", metadata.getSuccessCount());
+        System.out.printf(
+            "Number of failures during the importing: %s\n", metadata.getFailureCount());
+      }
+
+      if (operation.hasResponse()) {
+        ImportProductsResponse response =
+            operation.getResponse().unpack(ImportProductsResponse.class);
+        System.out.printf("Operation result: %s%n", response);
+      }
+    }
+  }
+
+  public static ImportProductsRequest getImportProductsBigQueryRequest(
+      ReconciliationMode reconciliationMode) {
+    BigQuerySource bigQuerySource =
+        BigQuerySource.newBuilder()
+            .setProjectId(PROJECT_ID)
+            .setDatasetId(DATASET_ID)
+            .setTableId(TABLE_ID)
+            .setDataSchema(DATA_SCHEMA)
+            .build();
+
+    ProductInputConfig inputConfig =
+        ProductInputConfig.newBuilder().setBigQuerySource(bigQuerySource).build();
+
+    ImportProductsRequest importRequest =
+        ImportProductsRequest.newBuilder()
+            .setParent(DEFAULT_CATALOG)
+            .setReconciliationModeValue(reconciliationMode.getNumber())
```

Comment:
Can we just setReconciliationMode(reconciliationMode)?

Reply:
fixed